diff -rauN file-5.33/src/seccomp.c file-5.33-seccomp-32bit-patch/src/seccomp.c
--- file-5.33/src/seccomp.c	2017-11-04 02:14:25.000000000 +0100
+++ file-5.33-seccomp-32bit-patch/src/seccomp.c	2018-05-03 13:38:38.073165287 +0200
@@ -168,13 +168,19 @@
 	ALLOW_RULE(dup2);
 	ALLOW_RULE(exit);
 	ALLOW_RULE(exit_group);
-	ALLOW_RULE(fcntl);  
+	ALLOW_RULE(fcntl);
+	ALLOW_RULE(fcntl64);  
 	ALLOW_RULE(fstat);
+	ALLOW_RULE(fstat64);
 	ALLOW_RULE(getdents);
+	ALLOW_RULE(getdents64);
 	ALLOW_RULE(ioctl);
 	ALLOW_RULE(lseek);
+	ALLOW_RULE(lseek64);
 	ALLOW_RULE(lstat);
+	ALLOW_RULE(lstat64);
 	ALLOW_RULE(mmap);
+	ALLOW_RULE(mmap2);
 	ALLOW_RULE(mprotect);
 	ALLOW_RULE(mremap);
 	ALLOW_RULE(munmap);
@@ -188,6 +194,7 @@
 	ALLOW_RULE(rt_sigreturn);
 	ALLOW_RULE(select);
 	ALLOW_RULE(stat);
+	ALLOW_RULE(stat64);
 	ALLOW_RULE(sysinfo);
 	ALLOW_RULE(unlink);
 	ALLOW_RULE(write);
