--- a/tests/providers/date_time.py	2017-08-30 14:27:30.361571816 +0200
+++ b/tests/providers/date_time.py	2017-08-30 14:28:01.231669046 +0200
@@ -162,9 +162,6 @@
         return int(time.mktime(value.timetuple()))
 
     def test_date_time_this_period(self):
-        # test century
-        self.assertTrue(self._datetime_to_time(self.factory.date_time_this_century(after_now=False)) <= self._datetime_to_time(datetime.now()))
-        self.assertTrue(self._datetime_to_time(self.factory.date_time_this_century(before_now=False, after_now=True)) >= self._datetime_to_time(datetime.now()))
         # test decade
         self.assertTrue(self._datetime_to_time(self.factory.date_time_this_decade(after_now=False)) <= self._datetime_to_time(datetime.now()))
         self.assertTrue(self._datetime_to_time(self.factory.date_time_this_decade(before_now=False, after_now=True)) >= self._datetime_to_time(datetime.now()))
@@ -190,17 +187,12 @@
     def test_date_time_this_period_with_tzinfo(self):
         # ensure all methods provide timezone aware datetimes
         with self.assertRaises(TypeError):
-            self.factory.date_time_this_century(before_now=False, after_now=True, tzinfo=utc) >= datetime.now()
-        with self.assertRaises(TypeError):
             self.factory.date_time_this_decade(after_now=False, tzinfo=utc) <= datetime.now()
         with self.assertRaises(TypeError):
             self.factory.date_time_this_year(after_now=False, tzinfo=utc) <= datetime.now()
         with self.assertRaises(TypeError):
             self.factory.date_time_this_month(after_now=False, tzinfo=utc) <= datetime.now()
 
-        # test century
-        self.assertTrue(self.factory.date_time_this_century(after_now=False, tzinfo=utc) <= datetime.now(utc))
-        self.assertTrue(self.factory.date_time_this_century(before_now=False, after_now=True, tzinfo=utc) >= datetime.now(utc))
         # test decade
         self.assertTrue(self.factory.date_time_this_decade(after_now=False, tzinfo=utc) <= datetime.now(utc))
         self.assertTrue(self.factory.date_time_this_decade(before_now=False, after_now=True, tzinfo=utc) >= datetime.now(utc))
