diff -rauN mplayer-38101/cpudetect.c mplayer-38101-sigcontext-patch/cpudetect.c
--- mplayer-38101/cpudetect.c	2018-05-17 11:15:24.975263641 +0200
+++ mplayer-38101-sigcontext-patch/cpudetect.c	2018-05-17 12:04:52.811930307 +0200
@@ -38,6 +38,7 @@
 #include <sys/types.h>
 #include <sys/sysctl.h>
 #elif defined(__linux__)
+#define __USE_MISC // for struct struct sigcontext
 #include <signal.h>
 #elif defined(__MINGW32__) || defined(__CYGWIN__)
 #include <windows.h>
